### PR TITLE
docs: add incident management and postmortem docs

### PR DESCRIPTION
**Pull request by monotykamary**
#### What's this PR do?

- [x] Adds details on incident management
- [x] Adds the 5 Whys technique to use for postmortems
- [x] Adds basic template outline for postmortems based on Atlassian work

### CODE REVIEW

File: engineering/incident-management.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+## Purpose
+
+In order to establish trust in our software or the delivery of our software, we need to be timely and transparent with how we manage incidents. Good incident management involves not only high reactivity to difficulties, but proper follow-ups to notify relevant parties of the severity of the issue, how we manage the problem, and ETA and steps to resolution of the issue.
+
+## Incident management
+
+The overall procedure for monitoring alerts and taking action includes determining an incident's possible impact and severity and identifying it as an issue and giving it a priority. If a problem condition cannot be found, we will reject it as a non-impacting event.
+
+### Reporting the problem
```
Comment (zlatanpham) [inline engineering/incident-management.md:16]: Do we need to mention we need to frequently do reporting? As the business and the customer need to know how long the problem has been resolved.

**Review by zlatanpham — COMMENTED** [review]
This looks good. Left a couple of suggestions.

File: engineering/incident-management.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+The simplest way to interrogate a problem is using the 5 Whys technique. [5 Whys](https://en.wikipedia.org/wiki/5_Whys) is an iterative interrogative technique developed by Sakichi Toyoda and is used to explore the cause-and-effect relationships underlying a problem. For example:
+
+- **Problem**: Our users have an issue viewing updates in tracking orders for their delivery.
+- **1st Why**: There is an issue with our view model.
+- **2nd Why**: Our view model has issues getting data from our closest database.
+- **3rd Why**: Our nearest database has connection issues egressing data.
+- **4th Why**: Our nearest database cannot resolve DNS properly to point and egress data.
+- **5th Why**: Our database has had a config change that affected the resolution of DNS queries.
+
+## Postmortems
```
[the remark this answers is not in the document]
Comment (monotykamary) [reply engineering/incident-management.md:60]: I don't agree it being a blame event though, did you mean blameless? 🤔

[the remark this answers is not in the document]
Comment (zlatanpham) [reply engineering/incident-management.md:51]: Sry. I've updated the messgae.

**Review by zlatanpham — APPROVED** [review]
Great work 👍

File: engineering/incident-management.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+#### TEMPLATE 
+- the action that mitigated the issue
+- why the action was taken
+- the outcome of the action
+
+### Timeline
+Detail the incident timeline with as much information as possible. Use UTC to standardize for timezones:
+
+### **TEMPLATE:**
```
Comment (vdhieu) [inline engineering/incident-management.md:113]: this should be 
```suggestion
#### **TEMPLATE:**
```

**Review by vdhieu — COMMENTED** [review]
@monotykamary just a small comment on the format
Response, Recovery, Timeline, and Root cause identification should be formatted as h2 right 🤔

**Comment by monotykamary** [comment]
> @monotykamary just a small comment on the format Response, Recovery, Timeline, and Root cause identification should be formatted as h2 right 🤔

My bad, they should be H3 as they are child topics of postmortems.

**Pull request merged**